nntrainer/opencl/opencl_program.cpp

```diff
@@ -41,6 +43,105 @@ bool Program::BuildProgram(cl_device_id device_id,
     return false;
   }
 
+  // saving kernel binary
+  if (!binaryCreated)
+    return GetProgramInfo(device_id);
+
+  return true;
+}
+
+/**
+ * @brief Utility to get program info and save kernel binaries
+ *
+ * @param device_id OpenCL device id
+ * @return true if successful or false otherwise
+ */
+bool Program::GetProgramInfo(cl_device_id device_id) {
+  // since only GPU is being used
+  unsigned int num_devices = 1;
+
+  cl_int error_code = CL_SUCCESS;
+  size_t *binaries_size = NULL;
+  unsigned char **binaries_ptr = NULL;
+  char *kernel_names;
+
+  // Read the binary size
+  size_t binaries_size_alloc_size = sizeof(size_t) * num_devices;
+  binaries_size = (size_t *)malloc(binaries_size_alloc_size);
+  error_code = clGetProgramInfo(program_, CL_PROGRAM_BINARY_SIZES,
+                                binaries_size_alloc_size, binaries_size, NULL);
```

Comment:
In C++, ```nullptr``` is proper for this pointer argument. Entering ```NULL``` may generate compiler errors in the future when the code gets extended.